src/components/CallToAction/index.tsx

```diff
@@ -0,0 +1,120 @@
+    href?: string
+    to?: string
+    displayBrackets?: boolean
+}
+
+const icons = {
+    rocket: rocketImg,
+    calendar: calendarImg,
+    github: githubImg,
+    handbook: handbookImg,
+    roadmap: roadmapImg,
+    check: checkImg,
+    book: bookImg,
+    none: null,
+}
+
+const buttonTypeClasses = {
+    secondary:
+        'bg-transparent border-2 border-white border-opacity-30 text-white hover:bg-white hover:bg-opacity-10 hover:text-white',
+    primary: 'bg-primary border-primary text-white hover:border-primary-dark hover:bg-primary-dark hover:text-white',
+    custom: '',
+}
+
+export const CallToAction = ({
+    className = '',
+    type = 'primary',
+    icon = 'none',
+    children,
+    width = '56',
+    href,
+    to,
+    onClick,
+    displayBrackets = false,
+}: CallToActionProps) => {
+    const iconNode = icons[icon] ? (
+        <div className="bg-opacity-10 bg-yellow-100 rounded rounded-sm p-1 mr-8">
+            <img src={icons[icon]} className="h-4 w-4 mb-0" alt="Get started with PostHog" />
+        </div>
+    ) : null
+
+    const brackets =
+        type === 'primary' || displayBrackets ? (
+            <>
+                <img
+                    src={buttonBracketImg}
+                    alt="button bracket"
+                    className="absolute top-0 left-0 -mt-2 -ml-2 opacity-30"
+                />
+                <img
+                    src={buttonBracketImg}
+                    alt="button bracket"
+                    className="absolute top-0 right-0 -mt-2 -mr-2 opacity-30 transform rotate-90"
+                />
+                <img
+                    src={buttonBracketImg}
+                    alt="button bracket"
+                    className="absolute bottom-0 left-0 -mb-2 -ml-2 opacity-30 transform -rotate-90"
+                />
+                <img
+                    src={buttonBracketImg}
+                    alt="button bracket"
+                    className="absolute bottom-0 right-0 -mb-2 -mr-2 opacity-30 transform rotate-180"
+                />
+            </>
+        ) : null
+
+    const widthClass = `w-${width}`
+    const baseClasses = `p-2 ${widthClass} uppercase rounded-sm flex items-center justify-between relative`
+    const classList = [baseClasses, buttonTypeClasses[type], className].join(' ')
+
+    const innerHtml = (
+        <>
+            {brackets}
+            {iconNode}
+            <div className="mr-8">{children}</div>
+            <span></span>
+        </>
+    )
+
+    if (href) {
+        return (
+            <a href={href} target="_blank" className={classList} rel="noreferrer">
+                {innerHtml}
+            </a>
+        )
+    } else if (to) {
+        return (
+            <Link to={to} className={classList}>
+                {innerHtml}
+            </Link>
+        )
+    } else {
+        return (
+            <button onClick={onClick} className={classList}>
+                {innerHtml}
+            </button>
+        )
+    }
```

Comment:
[non-blocking] nit - this could be:

```
return href ? (
            <a href={href} target="_blank" className={classList} rel="noreferrer">
                {innerHtml}
            </a>
        ) : to ? (
            <Link to={to} className={classList}>
                {innerHtml}
            </Link>
        ) : (
            <button onClick={onClick} className={classList}>
                {innerHtml}
            </button>
        )
```

Reply:
🤦‍♂️ It never dawned on me that you could do that. I think that's a bit cleaner than multiple explicit returns though!